cmd/serv.go

```diff
@@ -181,7 +181,14 @@ func runServ(c *cli.Context) error {
 
 	rr := strings.SplitN(repoPath, "/", 2)
 	if len(rr) != 2 {
-		return fail("Invalid repository path", "Invalid repository path: %v", repoPath)
+		if len(setting.SSH.RedirectPath) > 0 {
+			redirectPath := strings.ToLower(setting.SSH.RedirectPath)
+			// Fake repoPath and we know len of rr will be 2
+			repoPath = redirectPath + "/" + repoPath
+			rr = strings.SplitN(repoPath, "/", 2)
```

Comment:
```suggestion
			rr = []string{setting.SSH.RedirectPath, repoPath}
```

Reply:
The string repoPath is used further down in the gitcmd and needs to have the redirect username / organisation set to find the repo. So I don't think we can skip building a new repoPath string with the redirect prefix.

Reply:
```suggestion
			repoPath = strings.ToLower(setting.SSH.RedirectPath) + "/" + repoPath
			rr = []string{setting.SSH.RedirectPath, repoPath}
```